src/adagrams.js

```diff
@@ -1,8 +1,102 @@
+// externally-accessed functions
 const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
+  // hand of drawn letters
+  drawLetters(){
+    // Is reference for number of letters that will populate the bucket
+    const alphaRef = { a: 9, b: 2, c: 2, d: 4, e: 12, f: 2, g: 3, h: 2, i: 9, j: 1, k: 1, l: 4, m: 2, n: 6, o: 8, p: 2, q: 1, r: 6, s: 4, t: 6, u: 4, v: 2, w: 2, x: 1, y: 2, z: 1 };
+
+    // initially empty bucket
+    let bucket = [];
+
+    // iterates over array, defining the letter and value. It then pushes the letter into the bucket array according to the value
+    for ( const character in alphaRef ) {
+      let value = alphaRef[character];
+      let letter = character;
+      let i = 0;
+      while (i < value) {
+        bucket.push(letter.toUpperCase())
+        i++
+      };
+    };
+
+    // shuffles bucket array 
+    const shuffleArray = function (array) {
+      for (var i = array.length - 1; i > 0; i--) {
+          var j = Math.floor(Math.random() * (i + 1));
+          var temp = array[i];
+          array[i] = array[j];
+          array[j] = temp;
+      };
+    };
+
+    // calls shuffling function
+    shuffleArray(bucket);
+
+    // hand that will be populated by now shuffled bucket
+    let hand = [];
+
+    // populates hand without destroying / altering bucket
+    for (let i = 0; i < 10; i++) {
+      hand.push(bucket[i])
+    };
+    return hand
+  },
+
+  usesAvailableLetters(input, lettersInHand){
+    // evaluates if hand uses only available letters
+    
+    //creates object (handyObject) that will be used as a lookup hash equivalent with the values being the number of occurences for a sinle letter. Once populated, it iterates over the input, decrementing the values 
+    let handyObject = {};
+
+    for(let i = 0; i < lettersInHand.length; i++){
+      if(!Object.keys(handyObject).includes(lettersInHand[i])) {
+        let key = lettersInHand[i];
+        Object.assign(handyObject, {[key]: 1});
+      }
+      else {
+        handyObject[lettersInHand[i]] += 1; 
+      }
+    }
+
+    for(let i = 0; i < input.length; i++) {
```

Comment:
I suggest moving away from these old-fashioned for loops and toward forEach and other loops.